gitlab/resource_gitlab_project_approval_rule_test.go

```diff
@@ -0,0 +1,416 @@
+package gitlab
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var gitlabProjectApprovalRuleSkipAttributes = []string{
+	"project", "user_ids", "group_ids", "eligible_approvers",
+}
+
+type checkGitlabProjectApprovalRule struct {
+	SkippedAttributes []string
+}
+
+func (c *checkGitlabProjectApprovalRule) Aggregate(name string, expected gitlab.ProjectApprovalRule) resource.TestCheckFunc {
```

Comment:
These methods do not need to be exported (public). Also I see some more things that are exported and do not need to be -- the `SkippedAttributes` field, your methods on `memberList`, and fields of `gitlabProjectApprovalRuleFixtures`.